src/rust/engine/src/externs/interface.rs

```diff
@@ -1320,6 +1327,37 @@ fn merge_directories(
   })
 }
 
+fn digest_to_bytes(
+  py: Python,
+  scheduler_ptr: PyScheduler,
+  py_digest: PyObject,
+) -> CPyResult<PyBytes> {
+  with_scheduler(py, scheduler_ptr, |scheduler| {
+    let core = scheduler.core.clone();
+    let store = core.store().clone();
+
+    let digest = crate::nodes::lift_digest(&py_digest.into())
+      .map_err(|e| PyErr::new::<exc::Exception, _>(py, (e,)))?;
+
+    let bytes: Vec<u8> = py
+      .allow_threads(|| {
+        core.executor.block_on(async move {
+          store
+            .load_file_bytes_with(digest, |bytes| bytes.into()) //TODO see if this can be done no-copy
+            .await
+            .and_then(|maybe_bytes: Option<(Vec<u8>, _)>| {
+              maybe_bytes
+                .map(|bytes_tuple| bytes_tuple.0)
+                .ok_or_else(|| format!("Error loading bytes from digest: {:?}", digest))
+            })
+        })
+      })
+      .map_err(|e| PyErr::new::<exc::Exception, _>(py, (e,)))?;
+
+    Ok(PyBytes::new(py, &bytes))
```

Comment:
So, this will copy these bytes out of this Vec, and we will already have copied them out of the database.

What you could do instead here is copy out of the database directly into a PyBytes object. Which, handily, we have a method for!:
```
.load_file_bytes_with(digest, externs::store_bytes)
```
...which will give you back a `Value`, which you can call `into()` on to get a `PyObject`. Due to that API not returning `PyBytes`, you'd have to change this method's return type to `PyObject` (although it is still bytes underneath). But you'll h